atak-civ/plugin-examples/plugintemplate/app/src/main/java/com/atakmap/android/plugintemplate/CreateTimerDropDown.java

```diff
@@ -0,0 +1,176 @@
+                        !durationHrStr.matches("") && !durationSecStr.matches("")) {
+                    timer.setName(nameStr);
+                    timer.setDuration(durationHours.getText().toString() + ":" +
+                            durationMinutes.getText().toString() + ":" +
+                            durationSeconds.getText().toString());
+                    timer.setPreset(preset.isChecked());
+                    timer.setMinutes(Integer.parseInt(durationMinStr));
+                    timer.setHours(Integer.parseInt(durationHrStr));
+                    timer.setSeconds(Integer.parseInt(durationSecStr));
+                    Intent i = new Intent();
+                    i.setAction(PluginTemplateDropDownReceiver.SHOW_PLUGIN);
+                    i.putExtra("TIMER", timer);
+                    AtakBroadcast.getInstance().sendBroadcast(i);
+                }
+            }
+        });
+        Button back = (Button)templateView.findViewById(R.id.back);
+        back.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent i = new Intent();
+                i.setAction(PluginTemplateDropDownReceiver.SHOW_PLUGIN);
+                AtakBroadcast.getInstance().sendBroadcast(i);
+            }
+        });
+
+    }
+
+    /**************************** PUBLIC METHODS *****************************/
+
+    public void disposeImpl() {
+    }
+
+    /**************************** INHERITED METHODS *****************************/
+
+    @Override
+    public void onReceive(Context context, final Intent intent) {
+        final String action = intent.getAction();
+        if (action == null)
+            return;
+
+        if (action.equals(SHOW_CREATE)) {
+
+            Log.d(TAG, "showing plugin drop down");
+            showDropDown(templateView, HALF_WIDTH, FULL_HEIGHT, FULL_WIDTH,
+                    HALF_HEIGHT, false);
+
+            if (intent.getStringExtra("SELECTED_SOUND") != null) {
+                this.changeSoundButton.setText(intent.getStringExtra("SELECTED_SOUND"));
+                timer.setSound(intent.getStringExtra("SELECTED_SOUND"));
+            } else if(intent.getStringArrayListExtra("SELECTED_NOTIFICATIONS") !=null) {
+                ArrayList<String> act = (intent.getStringArrayListExtra("SELECTED_NOTIFICATIONS"));
+                timer.setNotification(act);
+            } else {
+                this.changeSoundButton.setText("Change");
+                this.name.setText("");
+                this.durationHours.setText("");
+                this.durationMinutes.setText("");
+                this.durationSeconds.setText("");
+                this.preset.setSelected(false);
+                this.timer.setSound("Chime");
+                this.changeSoundButton.setText("Chime");
+                this.timer.setNotification(new ArrayList(Arrays.asList("At Time of Event")));
```

Comment:
I would just change this to something like:
String defaultSound = pluginContext.getResources().getStringArray(R.array.custom_sounds)[0];
                this.timer.setSound(defaultSound);
                this.changeSoundButton.setText(defaultSound);
                String defaultNotification = pluginContext.getResources().getStringArray(R.array.custom_notification_settings)[0];
                this.timer.setNotification(new ArrayList(Arrays.asList(defaultNotification)));

so that the strings aren't hardcoded

Reply:
fixed